### PR TITLE
Pimcore 3 Upgrade

### PR DESCRIPTION
**Pull request by seeruk**
What is in this PR?
- MIT License applied
- Updated composer.json with Byng-owned Packagist package name that follows newly set out conventions, new license, homepage set to this repository
- Removed composer.lock, as this is a library
- Updated version of PHP Elasticsearch client
- Corrected spelling of Elasticsearch from Elastic Search to Elasticsearch
- Updated namespaces to be `Byng\Pimcore\` prefixed
- Consistent style in doc blocks, document headers, and code
- Slightly updated structure in some areas
- Pimcore 3 support (namespaces and updated plugin entry point and configuration)
- Renamed config distribution file to match the name it is copied to
- Added editor config configuration
- Removed coupling to any objects defined in Pimcore (I am planning on submitting another PR with support for extending this plugin somehow, it may be a 3.x.x version though as their may be some more fairly large BC breaks there)

### CODE REVIEW

**Comment by seeruk** [comment]
It is important to remember, as this package was never submitted to Packagist initially, other projects may be using dev-development as their version constraint and a custom repository field. Those projects will need to be updated to ensure they use the now-tagged v1.0.0 version, which will be available on Packagist when this is submitted.

File: composer.json [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-    "license": "BSD-3-Clause",
-    "description": "Elastic Search Pimcore plugin",
-    "keywords": ["pimcore", "elastic search", "plugin"],
-    "homepage": "http://byng.co",
+    "license": "MIT",
+    "keywords": [ "pimcore", "elasticsearch", "plugin" ],
+    "homepage": "https://github.com/byng-systems/pimcore-elastic-search-plugin",
     "authors": [
         {
             "name": "Byng Systems",
```
Comment (mdjward) [inline composer.json:11]: I really think we should start putting down specific names here (using the commit log as a reference).  Thoughts?

Comment (seeruk) [inline composer.json:11]: Agreed, I think that's a good idea. We can keep Byng *Services (which I need to change this to) at the top so people still have the Byng contact information too.

File: lib/Byng/Pimcore/ElasticSearch/Filter/TagKeyFilter.php [hunk 2]
```diff
@@ -1,101 +1,104 @@
...
     /**
-     * 
-     * @param array $replaceable
-     * @param type $replacement
+     * Constructor
+     *
+     * @param array|null $replaceable
+     * @param string|null $replacement
      */
     public function __construct(array $replaceable = null, $replacement = null)
```
Comment (seeruk) [inline lib/Byng/Pimcore/ElasticSearch/Filter/TagKeyFilter.php]: `$replaceable` could do with being defaulted to an empty array actually, so the null check wouldn't be necessary.

File: lib/Byng/Pimcore/ElasticSearch/Repository/PageRepository.php [hunk 3]
```diff
@@ -186,55 +185,55 @@ public function findBy(
         $additionalOptions = []
     ) {
         $body = $additionalOptions + [
-            'query' => [
-                'bool' => [
-                    'must' => $mustCriteria,
-                    'should' => $shouldCriteria,
-                    'must_not' => $mustNotCriteria
+            "query" => [
```
Comment (mdjward) [inline lib/Byng/Pimcore/ElasticSearch/Repository/PageRepository.php]: Any particular reason for the quotation changes?

Comment (seeruk) [inline lib/Byng/Pimcore/ElasticSearch/Repository/PageRepository.php]: Just for the sake of consistency throughout the codebase.

**Pull request merged**